services/venice-controller/src/main/java/com/linkedin/venice/controller/stats/LogCompactionStats.java

```diff
@@ -0,0 +1,134 @@
+package com.linkedin.venice.controller.stats;
+
+import static com.linkedin.venice.stats.dimensions.VeniceMetricsDimensions.VENICE_CLUSTER_NAME;
+
+import com.google.common.collect.ImmutableMap;
+import com.linkedin.venice.stats.AbstractVeniceStats;
+import com.linkedin.venice.stats.VeniceMetricsConfig;
+import com.linkedin.venice.stats.VeniceMetricsRepository;
+import com.linkedin.venice.stats.VeniceOpenTelemetryMetricsRepository;
+import com.linkedin.venice.stats.dimensions.RepushStoreTriggerSource;
+import com.linkedin.venice.stats.dimensions.VeniceMetricsDimensions;
+import com.linkedin.venice.stats.dimensions.VeniceResponseStatusCategory;
+import com.linkedin.venice.stats.metrics.MetricEntityStateGeneric;
+import com.linkedin.venice.stats.metrics.TehutiMetricNameEnum;
+import io.tehuti.metrics.MetricsRepository;
+import io.tehuti.metrics.stats.Count;
+import io.tehuti.metrics.stats.Gauge;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class LogCompactionStats extends AbstractVeniceStats {
+  private final boolean emitOpenTelemetryMetrics;
+  private final VeniceOpenTelemetryMetricsRepository otelRepository;
+  private final Map<VeniceMetricsDimensions, String> baseDimensionsMap;
+
+  /** metrics */
+  private final MetricEntityStateGeneric repushCallCountMetric;
+  private final MetricEntityStateGeneric compactionEligibleMetric;
+  private final MetricEntityStateGeneric storeNominatedForCompactionCountMetric;
+
+  public LogCompactionStats(MetricsRepository metricsRepository, String clusterName) {
+    super(metricsRepository, "LogCompactionStats");
+    if (metricsRepository instanceof VeniceMetricsRepository) {
+      VeniceMetricsRepository veniceMetricsRepository = (VeniceMetricsRepository) metricsRepository;
+      VeniceMetricsConfig veniceMetricsConfig = veniceMetricsRepository.getVeniceMetricsConfig();
+      emitOpenTelemetryMetrics = veniceMetricsConfig.emitOtelMetrics();
+      if (emitOpenTelemetryMetrics) {
+        this.otelRepository = veniceMetricsRepository.getOpenTelemetryMetricsRepository();
+        this.baseDimensionsMap = new HashMap<>();
+        this.baseDimensionsMap.put(VENICE_CLUSTER_NAME, clusterName);
+      } else {
+        this.otelRepository = null;
+        this.baseDimensionsMap = null;
+      }
+    } else {
+      this.emitOpenTelemetryMetrics = false;
+      this.otelRepository = null;
+      this.baseDimensionsMap = null;
+    }
+
+    repushCallCountMetric = MetricEntityStateGeneric.create(
+        ControllerMetricEntity.REPUSH_CALL_COUNT.getMetricEntity(),
+        otelRepository,
+        this::registerSensor,
+        ControllerTehutiMetricNameEnum.REPUSH_CALL_COUNT,
+        Collections.singletonList(new Count()),
```

Comment:
Lets make this `OccurrenceRate` as Count will be a monotonically increasing metric.